cmd/gh-classroom/shared/shared.go

```diff
@@ -109,30 +109,48 @@ func ListAcceptedAssignments(client api.RESTClient, assignmentID int, page int,
 	return acceptedAssignmentList, nil
 }
 
+type assignmentList struct {
+	assignments []classroom.AcceptedAssignment
+	Error       error
+}
+
 func ListAllAcceptedAssignments(client api.RESTClient, assignmentID int, perPage int) (classroom.AcceptedAssignmentList, error) {
-	var page = 1
-	response, err := classroom.GetAssignmentList(client, assignmentID, page, perPage)
+
+	//Calculate the number of go channels to create. We will assign 1 channel per page
+	//so we don't put too much pressure on the API all at once
+	theAssignment, err := classroom.GetAssignment(client, assignmentID)
 	if err != nil {
 		return classroom.AcceptedAssignmentList{}, err
 	}
-
-	if len(response) == 0 {
-		return classroom.AcceptedAssignmentList{}, nil
-	}
-
-	//keep calling getAssignmentList until we get them all
-	var nextList []classroom.AcceptedAssignment
-	for hasNext := true; hasNext; {
-		page += 1
-		nextList, err = classroom.GetAssignmentList(client, assignmentID, page, perPage)
-		if err != nil {
-			return classroom.AcceptedAssignmentList{}, err
+	numChannels := (theAssignment.Accepted / perPage)
```

Comment:
We can remove the conditional below if we use float division and the ceiling round function 😄 

```suggestion
	numChannels := math.Ceil(float64(assignment.Accepted) / perPage))
```